go-controller/pkg/ovn/controller/services/svc_template_var.go

```diff
@@ -215,3 +210,60 @@ func getTemplatesFromRulesTargets(rules []LBRule) TemplateMap {
 	}
 	return templates
 }
+
+// NodeIPTemplates maintains templates variables for many IP addresses per node,
+// creating them in the form NODEIP_IPv4_0, NODEIP_IPv4_1, NODEIP_IPv4_2, ...
+// if and when they are needed.
+type NodeIPsTemplates struct {
+	ipFamily  corev1.IPFamily
+	templates []*Template
+}
+
+func NewNodeIPsTemplates(ipFamily corev1.IPFamily) *NodeIPsTemplates {
+	return &NodeIPsTemplates{
+		ipFamily:  ipFamily,
+		templates: make([]*Template, 0),
+	}
+}
+
+// AddIP adds a template variable for the specified chassis and ip address.
+func (n *NodeIPsTemplates) AddIP(chassisID string, ip net.IP) {
+
+	var i int = -1
```

Comment:
I'm new to golang so please take this with a grain of salt..

It took me a while to decipher this.  Would it be more clear if we rewrite it as:
```
func (n *NodeIPsTemplates) AddIP(chassisID string, ip net.IP) {

	for _, template := range n.templates {
		_, ok := template.Value[chassisID]
		if !ok {
			template.Value[chassisID] = ip.String()
			return
		}
	}

	// NODEIP_IPvN_XXX is missing, creating it.
	newTemplate := makeTemplate(
		makeLBNodeIPTemplateNamePrefix(n.ipFamily) + fmt.Sprint(len(n.templates)),
	)

	// And initialize with chassisID value
	newTemplate.Value[chassisID] = ip.String()

	n.templates = append(n.templates, newTemplate)
}
```

Reply:
Make sense! going to clean this